jqassistant/structure.adoc

```diff
@@ -0,0 +1,118 @@
+[[structure:Default]]
+[role=group,includesConstraints="structure:RootPackage,structure:FrontendDependency,structure:JobsDependency,structure:ServiceDependency"]
+== Structural Rules
+
+The package `net.sf.sze` is used as the root package of the application.
+
+[[structure:RootPackage]]
+.Labels the package `net.sf.sze` as `Root`.
+[source,cypher,role=concept]
+----
+MATCH
+  (root:Package)
+WHERE
+  root.fqn = "net.sf.sze"
+SET
+  root:Root
+RETURN
+  root as RootPackage
+----
+
+[[structure:RootPackage]]
+.All Java types must be located within the root package or a sub-package thereof.
+[source,cypher,role=constraint,requiresConcepts="structure:RootPackage,maven:InternalFile"]
+----
+MATCH
+  (type:Type:Internal)
+WHERE NOT
+  (:Package:Root)-[:CONTAINS*]->(type)
+RETURN
+  type as TypeOutsideRootPackage
+----
+
+All packages which are directly contained in the root package are technical slices.
+
+[[structure:Slice]]
+.Adds the label `Slice` to all packages contained in the root package (of all non-test artifacts).
+[source,cypher,role=concept,requiresConcepts="structure:RootPackage"]
+----
+MATCH
+  (artifact:Artifact)-[:CONTAINS]->(root:Package:Root)-[:CONTAINS]->(slice:Package)
+WHERE
+   artifact.type <> "test-jar"
```

Comment:
Where is the artifact.type set. Can't find such information in the documentation.